src/main/java/teammates/ui/webapi/AdminExceptionTestAction.java

```diff
@@ -40,8 +40,8 @@ JsonResult execute() {
         if (error.equals(DeadlineExceededException.class.getSimpleName())) {
             throw new DeadlineExceededException("DeadlineExceededException testing");
         }
-        if (error.equals(DatastoreTimeoutException.class.getSimpleName())) {
-            throw new DatastoreTimeoutException("DatastoreTimeoutException testing");
+        if (error.equals(DatastoreException.class.getSimpleName())) {
+            throw new DatastoreException(4, "DatastoreException testing", "DEADLINE_EXCEEDED");
```

Comment:
Maybe we can assign a variable name instead of hardcoding the HTTP status code here.

Besides, according to documentation:

> DEADLINE_EXCEEDED: A deadline was exceeded on the server.

It seems to be a server-side error and therefore status code should be 5xx?

Reply:
> It seems to be a server-side error and therefore status code should be 5xx?

I perceived the code to be one of the canonical error codes [here](https://github.com/googleapis/googleapis/blob/master/google/rpc/code.proto). I'm not 100% sure whether it refers to the canonical or the HTTP status code (if it is, it would be `504 Gateway Timeout`). 'DEADLINE_EXCEEDED' has error code `4` which I'm assuming corresponds to a timeout exception.
```
  // The deadline expired before the operation could complete. For operations
  // that change the state of the system, this error may be returned
  // even if the operation has completed successfully.  For example, a
  // successful response from a server could have been delayed long
  // enough for the deadline to expire.
  //
  // HTTP Mapping: 504 Gateway Timeout
  DEADLINE_EXCEEDED = 4;
```

Reply:
@jianhandev OK, I think 4 should be correct.

Their devs seem to be confused as well. 
https://github.com/googleapis/google-cloud-java/issues/2442#issuecomment-332303649